htdocs/accountancy/journal/purchasesjournal.php

```diff
@@ -152,7 +153,7 @@
 		$obj = $db->fetch_object($result);
 
 		// Controls
-		$compta_soc = ($obj->code_compta_fournisseur != "") ? $obj->code_compta_fournisseur : $cptfour;
+		$compta_soc = (!empty($obj->code_compta_fournisseur)) ? $obj->code_compta_fournisseur : $cptfour;
```

Comment:
If code_compta_fournisseur = '000', the change will not have the same effect.
Someone is the past has changed the "if empty) into != "" to be able to manage such cases (I don't know why this is needed, but why not). Do you need to remove this feature ?